pycaret/time_series.py

```diff
@@ -2480,3 +2480,46 @@ def __init__(
     def _run_search(self, evaluate_candidates):
         """Search all candidates in param_grid"""
         evaluate_candidates(ParameterGrid(self.param_grid))
+
+class ForecastingRandomizedSearchCV(BaseGridSearch):
+    def __init__(
+        self,
+        forecaster,
+        cv,
+        param_distributions,
+        n_iter=10,
+        scoring=None,
+        n_jobs=None,
+        refit=True,
+        refit_metric: str='smape',
+        verbose=0,
+        random_state=None,
+        pre_dispatch="2*n_jobs",
+        error_score=np.nan,
+        return_train_score=False
+    ):
+        super(ForecastingRandomizedSearchCV, self).__init__(
+            forecaster=forecaster,
+            cv=cv,
+            n_jobs=n_jobs,
+            pre_dispatch=pre_dispatch,
+            refit=refit,
+            refit_metric=refit_metric,
+            scoring=scoring,
+            verbose=verbose,
+            error_score=error_score,
+            return_train_score=return_train_score
+        )
+        self.param_distributions = param_distributions
+        self.n_iter = n_iter
+        self.random_state = random_state
+
+    def _run_search(self, evaluate_candidates):
```

Comment:
the `evaluate_candidates` is callable?

Reply:
Yes. its defined in the GridSearch class.

Reply:
even though it is a "private" method, perhaps we could raise an error if `evaluate_candidates` is not callable

Reply:
Since we are controlling that, it should be ok. In any case, it is mimicking the sklearn and sktime implementation so we should be ok.

Reply:
https://github.com/scikit-learn/scikit-learn/blob/95119c13af77c76e150b753485c662b7c52a41a2/sklearn/model_selection/_search.py#L1625